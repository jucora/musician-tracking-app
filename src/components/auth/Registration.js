/* eslint no-console: ["error", { allow: ["warn", "error"] }] */
import React from 'react';
import axios from 'axios';
import PropType from 'prop-types';

export default class Registration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      passwordConfirmation: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    const { handleSuccessfulAuth } = this.props;
    const { email, password, passwordConfirmation } = this.state;

    e.preventDefault();
    axios
      .post(
        'http://localhost:3001/registrations',
        {
          user: {
            email,
            password,
            passwordConfirmation,
          },
        },
        { withCredentials: true }
      )
      .then((response) => {
        handleSuccessfulAuth(response.data);
      })
      .catch((error) => {
        console.error('Registration error', error);
      });
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { email, password, passwordConfirmation } = this.state;
    return (
      <div className="form">
        <form onSubmit={this.handleSubmit}>
          <input
            type="email"
            name="email"
            placeholder="Your Email"
            value={email}
            onChange={this.handleChange}
            required
          />

          <input
            type="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={this.handleChange}
            required
          />

          <input
            type="password"
            name="passwordConfirmation"
            placeholder="Password confirmation"
            value={passwordConfirmation}
            onChange={this.handleChange}
            required
          />

          <button type="submit">Register</button>
        </form>
      </div>
    );
  }
}

Registration.propTypes = {
  handleSuccessfulAuth: PropType.func.isRequired,
};
