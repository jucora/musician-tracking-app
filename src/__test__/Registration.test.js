import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Registration from '../components/auth/Registration';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(
    <Registration handleSuccessfulAuth={() => ''} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
