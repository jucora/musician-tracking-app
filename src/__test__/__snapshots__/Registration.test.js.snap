// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <form>
      <h2>
        Create Account
      </h2>
      <input
        name="email"
        placeholder="Your Email"
        required=""
        type="email"
        value=""
      />
      <input
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <input
        name="passwordConfirmation"
        placeholder="Password confirmation"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
